test/dag-node-test.js

```diff
@@ -32,459 +29,261 @@ module.exports = (repo) => {
+      expect(cid.version).to.equal(0)
+      const mh = multihash.decode(cid.multihash)
+      expect(mh.name).to.equal('sha2-256')
     })
 
-    it('get node CID with hashAlg', (done) => {
-      DAGNode.create(Buffer.from('some data'), (err, node) => {
-        expect(err).to.not.exist()
-        util.cid(node, { hashAlg: 'sha2-512' }, (err, cid) => {
-          expect(err).to.not.exist()
-          expect(cid.multihash).to.exist()
-          expect(cid.codec).to.equal('dag-pb')
-          expect(cid.version).to.equal(1)
-          const mh = multihash.decode(cid.multihash)
-          expect(mh.name).to.equal('sha2-512')
-          done()
-        })
-      })
+    it('get node CID with hashAlg', async () => {
+      const node = await DAGNode.create(Buffer.from('some data'))
+      const serialized = dagPB.util.serialize(node)
+      const cid = await dagPB.util.cid(serialized, { hashAlg: multicodec.SHA2_512 })
+      expect(cid.multihash).to.exist()
+      expect(cid.codec).to.equal('dag-pb')
+      expect(cid.version).to.equal(1)
+      const mh = multihash.decode(cid.multihash)
+      expect(mh.name).to.equal('sha2-512')
     })
 
-    it('marshal a node and store it with block-service', (done) => {
-      DAGNode.create(Buffer.from('some data'), (err, node) => {
-        expect(err).to.not.exist()
-        let block
-
-        waterfall([
-          (cb) => dagPB.util.serialize(node, cb),
-          (s, cb) => dagPB.util.cid(s, (err, cid) => {
-            cb(err, {
-              buffer: s,
-              cid: cid
-            })
-          }),
-          ({ buffer, cid }, cb) => {
-            block = new Block(buffer, cid)
-            bs.put(block, cb)
-          },
-          (cb) => bs.get(block.cid, cb),
-          (retrievedBlock, cb) => {
+    it('marshal a node and store it with block-service', async () => {
+      const node = await DAGNode.create(Buffer.from('some data'))
+      const serialized = dagPB.util.serialize(node)
+      const cid = await dagPB.util.cid(serialized)
+      const block = new Block(Buffer.from(serialized), cid)
+      return new Promise((resolve) => {
+        bs.put(block, (err) => {
+          expect(err).to.not.exist()
+          bs.get(block.cid, (err, retrievedBlock) => {
+            expect(err).to.not.exist()
             expect(retrievedBlock).to.eql(block)
-            cb()
-          }
-        ], done)
+            resolve()
+          })
+        })
       })
     })
 
-    it('deserialize go-ipfs block from ipldResolver', (done) => {
+    it('deserialize go-ipfs block from ipldResolver', async () => {
       if (!isNode) {
-        return done()
+        return
       }
 
       const cidStr = 'QmYwAPJzv5CZsnA625s3Xf2nemtYgPpHdWEz79ojWnPbdG'
       const cid = new CID(cidStr)
 
-      bs.get(cid, (err, block) => {
-        expect(err).to.not.exist()
-        dagPB.util.deserialize(block.data, (err, node) => {
+      return new Promise((resolve) => {
+        bs.get(cid, (err, block) => {
```

Comment:
if using the `promisify` wrapper, same thing applies here and you can ditch the error.to.not.exists and returns